src/software/control/src/display/drawer.rs

```diff
@@ -184,49 +180,39 @@ impl DisplayDrawer {
 
     // TODO: refactor, rename and relocate this
     fn add_pressure(&self, data: &mut DataPressure, new_point: u16) {
-        data.insert(0, (Local::now(), new_point / 10));
-        let oldest = data.first().unwrap().0 - Duration::seconds(40);
-        let newest = data.first().unwrap().0;
-        let mut i = 0;
-        while i != data.len() {
-            if oldest > (&mut data[i]).0 || newest < (&mut data[i]).0 {
-                let _ = data.remove(i);
-            } else {
-                i += 1;
+        let now = Local::now();
```

Comment:
Maybe we should use the `systick` field of `DataSnapshot` which is a timestamp (0 = when the MCU booted). _This can be done in a further PR._

Reply:
Oh indeed, didn't know there was a timestamp sent. Will do